content/_changelogs/10.7.0.md

```diff
@@ -0,0 +1,86 @@
+- Vue.js projects with a custom spec pattern configuration can now use the
+  "Create Spec from Component" capability introduced in v10.5.0. Addresses
+  [#23071](https://github.com/cypress-io/cypress/issues/23071).
+- Angular component testing projects can provide a custom configuration to the
+  Webpack dev server using the
+  [`projectConfig`](https://docs.cypress.io/guides/references/configuration#Options-API)
+  key. This enables support for projects that don't adhere to Angular CLI
+  conventions. Addresses
+  [#23161](https://github.com/cypress-io/cypress/issues/23161).
+- The Cypress `clock` can now be set to a new time using the `setSystemTime`
+  function. Addresses
+  [#15424](https://github.com/cypress-io/cypress/issues/15424).
+
+**Performance:**
+
+- The `cy.session()` command's behavior has been enhanced to no longer clear the
+  dom after a validation function runs. This means a `cy.visit()` command is no
+  longer required after running `cy.session()` when a validation function is
+  used. Addresses [#22368](https://github.com/cypress-io/cypress/issues/22368).
+
+**Bugfixes:**
+
+- Fixed an issue where Cypress could crash if a websocket upgrade request was
+  made over HTTPS. Fixes
+  [#22217](https://github.com/cypress-io/cypress/issues/22217).
+- Fixed an issue where filtering with the `--spec` CLI flag in run mode option
+  would not find any specs if the project was located at the root directory of a
+  file system. Fixes
+  [#23380](https://github.com/cypress-io/cypress/issues/23380).
+- Fixed an issue where projects using Angular `v14.2.0` and above would be
+  unable to run component tests. Fixes
+  [#23585](https://github.com/cypress-io/cypress/issues/23585).
+- Fixed an issue where snapshots performed in XHR requests could reflect the
+  primary domain instead of current domain. Fixes
+  [#21496](https://github.com/cypress-io/cypress/issues/21496).
+- An update was made to correctly track Github Actions retries in the Cypress
+  Dashboard. Previously retries data was not being recorded. Addressed in
+  [#23445](https://github.com/cypress-io/cypress/pull/23445).
+- Fixed an issue where an internal TypeScript type was exposed globally. Fixes
+  [#23333](https://github.com/cypress-io/cypress/pull/23333).
+- Fixed an issue where an incomplete
+  [`component.devServer`](https://docs.cypress.io/guides/references/configuration#devServer)
+  configuration would cause the specs list to not display in the Cypress App
+  when running Component Tests in open mode. Fixes
+  [#23486](https://github.com/cypress-io/cypress/issues/23486).
+
+**Miscellaneous:**
+
+- Display a default icon in the browser selection menu for browsers without
+  bundled icons. Fixes
+  [#21766](https://github.com/cypress-io/cypress/issues/21766).
+
+**Dependency Updates:**
+
+- Upgraded the bundled TypeScript version shipped with Cypress from `4.2.3` to
+  `4.7.4`. Addressed in
+  [#23523](https://github.com/cypress-io/cypress/pull/23523).
```

Comment:
@rachelruderman mentioned this [in a comment](https://github.com/cypress-io/cypress-documentation/pull/4697#discussion_r957874422) but it might have gotten lost amongst the changes. This seems like it was an internal dependency bump and not user-facing, I don't think it needs to be in the changelog.

Reply:
This did bump in @packages/server & webpack-dev-server so I do think this is user-facing -- it'll impact projusts using typescript right?

Reply:
It could impact the types we distribute, but it's also 1. a change in minor version and 2. currently unpinned in those projects and will change again without our knowing.

I guess my thought is that if this is passive to users, it's not worth calling out. And if it's not passive, we would need to look into [versioning our types](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#version-selection-with-typesversions).

Reply:
oh I was looking at the MVB of ts-loader: https://github.com/cypress-io/cypress/pull/23523/files#diff-92fadee1dfb23b20d6b0f6557a3ea0b8a231a7591db73f9f37772383bd0575d9R184

Reply:
It could def affect the types we ship if we utilize types that are only available in newer versions of TS (like the `Awaited` utility type). Until we utilize these types, we should be ok with the bump.

@tbiethman make sense to create an issue in regards to better handling of our TS types?

Reply:
For reference, we currently document our minimum supported TS version as 3.4: https://docs.cypress.io/guides/tooling/typescript-support#Install-TypeScript

To cover our bases, it appears we'd need to:
1. version our types (where necessary) so that we continue to support our documented minimum spec
2. reset expectations for the minimum spec with each major Cypress release to drop support as needed

I'll get an issue logged to explore this more.